src/main/java/chess/controller/ChessWebController.java

```diff
@@ -0,0 +1,176 @@
+            model.put("id", id);
+            return render(model, "chessGame.html");
+        });
+
+        post("/move", (req, res) -> {
+            String message = null;
+            Map<String, Object> model = new HashMap<>();
+            String number = req.queryParams("id").trim();
+            Long id = Long.valueOf(number);
+            List<String> parameters = new ArrayList<>(Arrays.asList(req.queryParams("parameter").split("_")));
+            try {
+                chessService.move(id, parameters);
+            } catch (IllegalArgumentException | UnsupportedOperationException e) {
+                model.put("error-message", e.getMessage());
+            }
+            ResponseDto responseDto = chessService.getResponseDto(id);
+            List<WebDto> boardDto = getBoardDto(responseDto.getBoard());
+            List<WebDto> scoreDto = getScoreDto(responseDto.getScores());
+            WebDto winnerDto = getTurnDto(responseDto.getWinner());
+            WebDto turnDto = getTurnDto(responseDto.getTurn());
+            model.put("board", boardDto);
+            model.put("score", scoreDto);
+            model.put("turn", turnDto);
+            model.put("message", responseDto.getMessage());
+            model.put("winner", winnerDto);
+            model.put("id", id);
+            return render(model, "chessGame.html");
+        });
+
+        post("/save", (req, res) -> {
+            String number = req.queryParams("id").trim();
+            Long id = Long.valueOf(number);
+            chessService.save(id);
+            List<WebDto> roomDto = getRoomDto(chessService.getRoomId());
+            Map<String, Object> model = new HashMap<>();
+            model.put("roomId", roomDto);
+            return render(model, "index.html");
+        });
+
+        post("/end", (req, res) -> {
+            String number = req.queryParams("id").trim();
+            Long id = Long.valueOf(number);
+            chessService.remove(id);
+            List<WebDto> roomDto = getRoomDto(chessService.getRoomId());
+            Map<String, Object> model = new HashMap<>();
+            model.put("roomId", roomDto);
+            return render(model, "index.html");
+        });
+
+        get("/", (req, res) -> {
+            List<WebDto> roomDto = getRoomDto(chessService.getRoomId());
+            Map<String, Object> model = new HashMap<>();
+            model.put("roomId", roomDto);
+            return render(model, "index.html");
+        });
+
+        get("/movable/:id/:position", (req, res) -> {
+            Long id = Long.valueOf(req.params(":id"));
+            Position position = Position.of(req.params(":position"));
+            List<Position> positions = chessService.getMovablePositions(id, position);
+            return positions.stream().map(Position::getName).collect(Collectors.joining(","));
+        });
+
+    }
+
+    private List<WebDto> getRoomDto(List<Long> roomIds) {
+        if (Objects.isNull(roomIds)) {
+            return null;
```

Comment:
null이 리턴될 가능성이 있을때 null로 리턴하는 것 보다 `Collections.emptyList();` 또는 `Optional`로 감싸서 리턴하면 NullPointerException을 방지 할 수 있어요.
```suggestion
            return Collections.emptyList();
```

Reply:
이 부분은 수정 했습니다!